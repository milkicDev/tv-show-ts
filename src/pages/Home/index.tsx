import React from 'react';
import { withTranslation } from 'react-i18next';
import ShowsApi from '../../api/Shows';
import ImageCard from '../../components/ImageCard';
import Schedule from '../../models/Schedule';

type Props = {
  t: Function;
};
type State = {
  data: Array<Schedule>,
};

class Home extends React.Component<Props, State> {
  state = {
    data: [],
  };

  componentDidMount() {
    ShowsApi.fetchAll()
      .then(res => {
        this.setState({ data: res });
      })
      .catch(error => console.error(error));
  }

  render() {
    const { t } = this.props;

    return (
      <div>
        <div className='header offset-bottom'>
          <div className='container'>
            <p>
              {t('home.title')}<br />
              <span dangerouslySetInnerHTML={{__html: t('home.description')}} />
            </p>

            <h2 className='offset-top'>Last Added Shows</h2>
          </div>
        </div>

        <div className='container offset--top'>
          <div className='flex-grid'>
            {this.state.data.map((item: Schedule, iItem: number) => (
              <ImageCard key={iItem} data={item.show} />
            ))}
          </div>
        </div>
      </div>
    );
  }
}

export default withTranslation('common')(Home);
