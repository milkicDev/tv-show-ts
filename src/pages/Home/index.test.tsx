import React from 'react';
import renderer from 'react-test-renderer'
import Home from '.';

it('render app', () => {
  const render = renderer.create(<Home />).toJSON();
  expect(render).toMatchSnapshot();
})
