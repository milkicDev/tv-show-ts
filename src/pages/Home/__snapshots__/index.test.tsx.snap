// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render app 1`] = `
<div>
  <div
    className="header offset-bottom"
  >
    <div
      className="container"
    >
      <p>
        home.title
        <br />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "home.description",
            }
          }
        />
      </p>
      <h2
        className="offset-top"
      >
        Last Added Shows
      </h2>
    </div>
  </div>
  <div
    className="container offset--top"
  >
    <div
      className="flex-grid"
    />
  </div>
</div>
`;
